### PR TITLE
Hard remove Razorback, old Reef, old Starlight

### PR DESCRIPTION
**Pull request by KingRaptor**
Also fixes Starlight referencing in lang files

See discussion on b0bbcbcfe4dac927a0369440947714955331d177

### CODE REVIEW

**Comment by Anarchid** [comment]
New!Starlight  is actually](https://github.com/ZeroK-RTS/Zero-K/blob/master/units/starlight.lua#L350 ) `mahlazer` for mission compatibility purposes. IIRC those are now invalid, so may as well rename to whatever is proper new naming scheme. `superstarlight`, `superchisel`, or `utilitystarlight`?

**Comment by GoogleFrog** [comment]
Fix the Starlight rule-breaking-naming.

File: units/starlight.lua [hunk 1]
```diff
@@ -347,4 +347,4 @@ unitDef = {
 }
 
 
-return lowerkeys({ mahlazer = unitDef})
+return lowerkeys({ starlight = unitDef})
```
Comment (GoogleFrog) [inline units/starlight.lua]: Doesn't this break everything?

Comment (KingRaptor) [inline units/starlight.lua]: I parsed "Fix the Starlight rule-breaking-naming" as requesting exactly this change, was it supposed to mean [this](https://github.com/ZeroK-RTS/Zero-K/issues/1596#issuecomment-267147718)?

I forgot that buildlists, etc. also need changing. Will revert, may as well only touch it once we actually do the consistent unitdef naming for everything at once.

Comment (sprunk) [inline units/starlight.lua]: I'd parse the request as to do the reverse (rename the file to mahlazer.lua).

Comment (Anarchid) [inline units/starlight.lua]: I think it makes sense to interpret it this way. This is because `starlight` fits one of the proposed rename schemes (proper name) while `mahlazer` has no chance of fitting any of them.

Comment (GoogleFrog) [inline units/starlight.lua]: The request does not have a specific way of fulfilling it. There is just a requirement to not break things.

File: units/mahlazer.lua [hunk 2]
```diff
@@ -1,5 +1,5 @@
 unitDef = {
-  unitname                      = [[mahlazer]],
+  unitname                      = [[starlight]],
```
Comment (GoogleFrog) [inline units/mahlazer.lua]: Did you run this to see if it works?

File: units/mahlazer.lua [hunk 3]
```diff
@@ -11,13 +11,13 @@ unitDef = {
   buildingGroundDecalDecaySpeed = 30,
   buildingGroundDecalSizeX      = 11,
   buildingGroundDecalSizeY      = 11,
-  buildingGroundDecalType       = [[mahlazer_aoplane.dds]],
-  buildPic                      = [[mahlazer.png]],
+  buildingGroundDecalType       = [[starlight_aoplate.dds]],
+  buildPic                      = [[starlight.png]],
```
Comment (GoogleFrog) [inline units/mahlazer.lua]: Breaks rules.

File: units/mahlazer.lua (hunk 2 above)
Comment (KingRaptor) [inline units/mahlazer.lua]: Yes.

(The unitname being `[[starlight]]` was already in the starlight.lua that's used in current stable and dev, it just looks like a change due to the git rename)

Comment (GoogleFrog) [inline units/mahlazer.lua]: Regardless, it looks wrong to me.

Comment (sprunk) [inline units/mahlazer.lua]: The value matters not to engine - there is no "unitname" unit def tag.

The value matters not to humans - it is a duplicate of the file name.
The tag should universally be removed in a separate commit.

**Comment by GoogleFrog** [comment]
Clippy conflict. The PR tries to do too much at once now.
